kernel/mempool.c

```diff
@@ -172,14 +172,25 @@ void k_free(void *ptr)
 
 #if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
 
+#if (CONFIG_HEAP_MEM_POOL_SIZE < CONFIG_HEAP_MEM_POOL_MIN_SIZE)
+#error "The smallest object must have a lower size then the entire heap"
+#endif
+#if (CONFIG_HEAP_MEM_POOL_MIN_SIZE & (CONFIG_HEAP_MEM_POOL_MIN_SIZE - 1)) != 0
+#error "The smallest object size must be a power of two"
+#endif
+#if (CONFIG_HEAP_MEM_POOL_SIZE & (CONFIG_HEAP_MEM_POOL_SIZE - 1)) != 0
+#error "The size of the heap must be a power of two"
```

Comment:
Same comment.  You can ask for a heap with a size of 65537 bytes, or a prime numer, and things will work, you'll just waste some memory.